SU2_CFD/include/numerics/NEMO/CNEMONumerics.hpp

```diff
@@ -70,6 +71,18 @@ class CNEMONumerics : public CNumerics {
                 unsigned short val_nPrimVarGrad,
                 const CConfig* config);
 
+  /*!
+   * \brief Constructor of the class, for turbulence applications.
+   * \param[in] val_nDim - Number of dimensions of the problem.
+   * \param[in] val_nVar - Number of variables of the problem.
+   * \param[in] config - Definition of the particular problem.
+   */
+  CNEMONumerics(unsigned short val_nDim, unsigned short val_nVar,
+                unsigned short val_nVar_NEMO,
```

Comment:
I think we don't need val_nVar_NEMO. val_nVar is a generic variable that stores the number of conservative variables for a given solver. I don't think there is the need to make it specific for NEMO and the way it is now, it seems simply like a repetition of val_nVar

Reply:
since we're here, we could also update the functions arguments in the doxy description of the function, to include val_nPrimVar, val_nPrimVarGrad

Reply:
Good catch, Ill update the Doxy stuff.    The reason there is a second CNEMONumerics is for use in the instantiation of the NEMO turb solver, I can remove for this PR (Maybe we can chat regarding a better way to handle the issue I am having when the time comes)

Reply:
ok, I didn't understand this was a new constructor, I thought it was replacing the old one. but still, if val_nVar = val_nVar_NEMO, what is the reasoning here ? if there's a logical explanation, of course we can keep it, just trying to understand

Reply:
So this constructor will be instantiated with turbulence solver where nVar = 1 (or 2).  Using the current numerics structure, the number of variable vs. vector size would be mismatched. The turbulence solution with be of size 1 or 2, but the nVar loop would be 6 or 7.  But we still need the index information which to do some computations.  So instead of trying to pass that information every time an index is needed, Instead pass the number of turbulence vars (val_nVar) and the number of Nemo vars.  I hope that makes sense

That being said, there may be way easier/efficient to do this.

Reply:
that makes total sense @WallyMaier . now that I understand better it actually sounds like a good and simple approach.